### PR TITLE
Add location hint APIs to Edge Network

### PR DESCRIPTION
**Pull request by cacheung**
Add setLocationHint and getLocationHint to Edge Network extension (ts, Java and iOS)
Add unit tests
Update Sample App with location hint API
Update Sample App AndroidManifest file's deeplink format, so now it works for all Android API.
Update location hint API doc.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -15,8 +15,10 @@
 
         <category android:name="android.intent.category.DEFAULT" />
         <category android:name="android.intent.category.LAUNCHER" />
-        <category android:name="android.intent.category.BROWSABLE" />
 
+      </intent-filter>
```
Comment (yangyansong-adbe) [inline apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml:19]: Can we revert this change? Do we have a reason to have two `<intent-filter>` sections?

Comment (cacheung) [reply apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml:19]: we need to have this change for the app to run on API 33.  The current setup gets error when running on API 33.  With this change, the app can be run on all Android API versions.

Comment (yangyansong-adbe) [reply apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml:19]: But, we already have the `<category android:name="android.intent.category.BROWSABLE" />` configuration. Why should we remove it and add it in another `<intent-filter>` section?

Comment (cacheung) [reply apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml:19]: It is about we need to move Deeplink setup into another intent-filter section to resolve the error when running on an API33 device. If not, we will get an error.
Error type 3
Error: Activity class does not exist. 

And while moving 
<data android:scheme="assurance" /> 
into another <intent-filter> section, it gives format error for <category android:name="android.intent.category.BROWSABLE" /> 
I need to move <category android:name="android.intent.category.BROWSABLE" /> into the next <intent-filter> to resolve the format error.

To keep the setup similar to the original AndroidManifest file and also work with API 33, the current code was what I tried and worked.

Comment (yangyansong-adbe) [reply apps/AEPSampleApp/android/app/src/main/AndroidManifest.xml:19]: Can we remove the `<data android:scheme="assurance" />` and see if it works with one `<intent-filter>`? I don't think we need the above "assurance" configuration for the sample app.

File: packages/edge/__tests__/EdgeTests.ts [hunk 2]
```diff
@@ -105,4 +105,23 @@ describe('Edge', () => {
...
+    expect(spy).toHaveBeenCalledWith('or2');
+  });
+
+  it('setLocationHint is called with null', async () => {
+    const spy = jest.spyOn(NativeModules.AEPEdge, 'setLocationHint');
+    Edge.setLocationHint(null);
+    expect(spy).toHaveBeenCalledWith(null);
+  });
+
+  it('getLocationHint is called', async () => {
```
[the remark this answers is not in the document]
Comment (cacheung) [reply packages/edge/__tests__/EdgeTests.ts:127]: It seems to be I can only test getLocationHint one time for the current setup (test/setup.ts).  I set the test to null and it returns null.  But I am not able to setup promise more than 1 time.  So I only leave the valid string test for getLocationHint right now.

**Review by kevinlind — APPROVED** [review]

**Review by yangyansong-adbe — APPROVED** [review]
